compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -2302,9 +2312,12 @@ public void visit(BLangQueryExpr queryExpr) {
         boolean prevWithinQuery = withinQuery;
         // This can be set to true directly since it's invoked only for nested queries.
         this.withinQuery = true;
+        boolean prevAfterGroupBy = this.afterGroupBy;
```

Comment:
We need to have a few test to include multiple group by in same query to test these lines

Reply:
Do you mean something like that,
https://github.com/ballerina-platform/ballerina-lang/pull/40287/files#diff-f33fd96d748c9aa10cdf8502de9966c2e338ef4ee796452e8c2ce5ca02d8c67fR777

Reply:
I will introduce it in separate PR